File: CODIGOS5/173.r
determine_chord_name <- function(notes) {
  steps <- diff(notes)
  steps <- (steps + 12) %% 12

  if (3 %in% steps & 4 %in% steps) {
    return("other")
  } else if (4 %in% steps) {
    return(get_chord_name(notes[1], TRUE))
  } else if (3 %in% steps) {
    return(get_chord_name(notes[1], FALSE))
  } else {
    return("other")
  }
}

get_chord_name <- function(root_note, is_major) {
  note_names <- c("C", "C#", "D", "D#", "E", "F", "F#", "G", "G#", "A", "A#", "B")
  root_note_name <- note_names[root_note %% 12 + 1]
  if (is_major) {
    return(paste0(root_note_name, "-major"))
  }
  return(paste0(root_note_name, "-minor"))
}

main <- function() {
  num_test_cases <- as.integer(readline())
  
  for (i in 1:num_test_cases) {
    input_notes <- as.integer(strsplit(readline(), " ")[[1]])
    chord_name <- determine_chord_name(input_notes)
    cat(chord_name, " ")
  }
}

main()

File: CODIGOS5/173.cs
using System;
using System.Collections.Generic;

class Program
{
    static void Main()
    {
        int numTestCases = int.Parse(Console.ReadLine());

        for (int i = 0; i < numTestCases; i++)
        {
            string[] inputNotes = Console.ReadLine().Split();
            int[] notes = Array.ConvertAll(inputNotes, int.Parse);

            string chordName = DetermineChordName(notes);
            Console.Write(chordName + " ");
        }
    }

    static string DetermineChordName(int[] notes)
    {
        List<int> steps = new List<int>();
        for (int i = 1; i < notes.Length; i++)
        {
            int diff = (notes[i] - notes[i - 1] + 12) % 12;
            steps.Add(diff);
        }

        if (steps.Contains(3) && steps.Contains(4))
        {
            return "other";
        }
        else if (steps.Contains(4))
        {
            return GetChordName(notes[0], true);
        }
        else if (steps.Contains(3))
        {
            return GetChordName(notes[0], false);
        }
        else
        {
            return "other";
        }
    }

    static string GetChordName(int rootNote, bool isMajor)
    {
        string[] noteNames = { "C", "C#", "D", "D#", "E", "F", "F#", "G", "G#", "A", "A#", "B" };
        string rootNoteName = noteNames[rootNote % 12];
        return rootNoteName + (isMajor ? "-major" : "-minor");
    }
}
